packages/flutter/lib/src/widgets/drag_target.dart

```diff
@@ -713,6 +713,7 @@ class _DragTargetState<T extends Object> extends State<DragTarget<T>> {
                                     (widget.onWillAccept != null &&
                                     widget.onWillAccept!(avatar.data as T?)) ||
                                     (widget.onWillAcceptWithDetails != null &&
+                                    avatar.data != null &&
                                     widget.onWillAcceptWithDetails!(DragTargetDetails<T>(data: avatar.data! as T, offset: avatar._lastOffset!)));
```

Comment:
I thought of doing the same check of `onWillAccept` for consistency

```suggestion
                                    (widget.onWillAccept != null &&
                                    avatar.data != null &&
                                    widget.onWillAccept!(avatar.data as T?)) ||
                                    (widget.onWillAcceptWithDetails != null &&
                                    avatar.data != null &&
                                    widget.onWillAcceptWithDetails!(DragTargetDetails<T>(data: avatar.data! as T, offset: avatar._lastOffset!)));
```

But doing so would force me to update this test: 

https://github.com/flutter/flutter/blob/127c76b91bef46c6baffaaee8873870c3054ea02/packages/flutter/test/widgets/draggable_test.dart#L3051-L3092

The last line would become:

```diff
-    expect(isReceiveNullDataForCheck, true);
+    expect(isReceiveNullDataForCheck, false); 
```

Reply:
I traced that test back to https://github.com/flutter/flutter/pull/72512

From your suggested code above:
```dart
// ...
avatar.data != null &&
widget.onWillAccept!(avatar.data as T?)) ||
// ...
```

^ This makes me wonder why would it then be cast as nullable `T?` if it is not null?

Having data be nullable appears to be intentional for onWillAccept, but that is being deprecated in https://github.com/flutter/flutter/pull/133691 so I think it makes sense to focus this change on onWillAcceptWithDetails.